app/src/main/java/de/tu_darmstadt/seemoo/nfcgate/nfc/NfcManager.java

```diff
@@ -318,7 +318,7 @@ private void enableDisableReaderMode() {
     private void enableForegroundDispatch() {
         Intent intent = new Intent(mActivity, mActivity.getClass());
         PendingIntent pendingIntent =
-                PendingIntent.getActivity(mActivity, 0, intent, 0);
+                PendingIntent.getActivity(mActivity, 0, intent, PendingIntent.FLAG_IMMUTABLE);
```

Comment:
Must be `MUTABLE` for foreground dispatch. I tested it on Android 14 with immutable and I received broadcasts with no actions. See also https://developer.android.com/develop/connectivity/nfc/advanced-nfc#foreground-dispatch
```suggestion
                PendingIntent.getActivity(mActivity, 0, intent, PendingIntent.FLAG_MUTABLE);
```